crates/core/tedge_actors/src/message_boxes.rs

```diff
@@ -0,0 +1,265 @@
+use crate::Builder;
+use crate::ChannelError;
+use crate::DynSender;
+use crate::Message;
+use crate::MessageBoxConnector;
+use crate::SimpleMessageBoxBuilder;
+use async_trait::async_trait;
+use futures::channel::mpsc;
+use futures::StreamExt;
+use log::debug;
+use log::info;
+
+/// A message box used by an actor to collect all its input and forward its output
+///
+/// This message box can be seen as two streams of messages,
+/// - inputs sent to the actor and stored in the message box awaiting to be processed,
+/// - outputs sent by the actor and forwarded to the message boxes of other actors.
+///
+/// ```logical-view
+///                                      +------+
+/// input_sender: DynSender<Input> ----->| Box  |----> output_sender: DynSender<Input>
```

Comment:
```suggestion
/// input_sender: DynSender<Input> ----->| Box  |<---- output_sender: DynSender<Output>
```

Reply:
The output type is indeed incorrect, but the arrow is.

```suggestion
/// input_sender: DynSender<Input> ----->| Box  |----> output_sender: DynSender<Output>
```

Reply:
This needs to be improved with two perspectives:

* from the actor point of view that consumes the input and produces the output,
* from the point of view of the peers that produces the input and consumes the output.

Reply:
Resolved as outdated.